driver/main.c

```diff
@@ -296,6 +302,189 @@ static void compat_unregister_trace(void *func, const char *probename, struct tr
 #endif
 }
 
+static int compat_set_tracepoint(void *func, const char *probename, struct tracepoint *tp, bool enabled)
+{
+	int ret = 0;
+	if (enabled)
+	{
+		ret = compat_register_trace(func, probename, tp);
+	}
+	else
+	{
+		compat_unregister_trace(func, probename, tp);
+	}
+	return ret;
+}
+
+static int force_tp_set(struct ppm_consumer_t *consumer, u32 new_tp_set, u32 max_val)
+{
+	u32 idx;
+	u32 new_val;
+	u32 curr_val;
+	int cpu;
+	int ret;
+	u32 stored_tp_set;
+
+	ret = 0;
+	stored_tp_set = g_tracepoints_attached;
+	for(idx = 0; idx < max_val && ret == 0; idx++)
+	{
+		new_val = new_tp_set & (1 << idx);
+		curr_val = g_tracepoints_attached & (1 << idx);
+
+		if (new_val)
+		{
+			// If enable is requested, set ref bit
+			g_tracepoints_refs[idx] |= 1 << consumer->id;
+		}
+		else
+		{
+			// If disable is requested, unset ref bit
+			g_tracepoints_refs[idx] &= ~(1 << consumer->id);
+		}
+
+		if(new_val == curr_val)
+		{
+			// no change needed, we just update the refs
+			continue;
+		}
+
+		if (g_tracepoints_refs[idx] != (1 << consumer->id) && g_tracepoints_refs[idx] != 0)
```

Comment:
Skip if:
* the tracepoint was already being requested by some other consumer
* the tracepoint is still requested by some other consumer

Otherwise go on and try to enable/disable the tracepoint.